asf_heat_pump_suitability/pipeline/prepare_features/listed_buildings.py

```diff
@@ -29,56 +29,95 @@ def transform_df_EPC_X_and_Y_to_point(
     Returns:
-        gpd.GeoDataFrame: EPC dataset with listed buildings grade without geometry.
+        pl.DataFrame: EPC dataset with listed buildings grade without geometry.
     """
-    epc_gdf = transform_df_EPC_X_and_Y_to_point(enhanced_epc_df)
-    epc_gdf_temp = epc_gdf[["geometry"]].copy()
-    epc_gdf_temp["index"] = epc_gdf.index
-    joined_gdf = gpd.sjoin(
-        epc_gdf_temp, listed_buildings_df, how="left", predicate="intersects"
+
+    # Add a unique index column for merging later
+    enhanced_epc_df = enhanced_epc_df.with_row_index("index")
+    partitions = enhanced_epc_df.with_row_count("chunk_id").select(
+        pl.col("chunk_id") // chunk_size
     )
-    # Drop the geometry column
-    joined_gdf = joined_gdf.drop(columns=["geometry", "index_right"])
-    epc_gdf["index"] = epc_gdf.index
-    result_gdf = epc_gdf.merge(
-        joined_gdf[["Grade", "index"]], on="index", how="left"
-    ).drop(columns=["geometry", "index"])
-    return result_gdf
+    # group based on the created index, resulting chunk_size partitons
+    data_partitioned = enhanced_epc_df.with_columns(partitions).partition_by("chunk_id")
+    logging.info(f"Adding listed buildings to EPC in {len(data_partitioned)} chunks")
+    for i, enhanced_epc_df_chunk in tqdm(enumerate(data_partitioned)):
+        epc_gdf = transform_df_EPC_X_and_Y_to_point(enhanced_epc_df_chunk)
+        epc_gdf_temp = epc_gdf[["geometry", "index"]].copy()
+        joined_gdf_chunk = gpd.sjoin(
+            epc_gdf_temp, listed_buildings_df, how="inner", predicate="intersects"
+        )
+        # Drop the geometry column
+        joined_gdf_chunk = joined_gdf_chunk.drop(columns=["geometry", "index_right"])
+        if i == 0:
+            joined_gdf = joined_gdf_chunk
+        else:
+            joined_gdf = pd.concat([joined_gdf, joined_gdf_chunk])
 
+    enhanced_epc_df = enhanced_epc_df.join(
+        pl.from_pandas(joined_gdf), how="left", on="index"
+    )
 
-def convert_gpd_to_polars(gpd: gpd.GeoDataFrame) -> pl.DataFrame:
-    """
-    Convert GeoDataFrame to Polars DataFrame.
+    return enhanced_epc_df
 
-    Args:
-        gpd (gpd.GeoDataFrame): GeoDataFrame to convert
 
+def merge_listed_buildings_nations(
+    enhanced_epc_nationone_listed_buildings_df: pl.DataFrame,
+    enhanced_epc_nationtwo_listed_buildings_df: pl.DataFrame,
+) -> pl.DataFrame:
+    """
+    Merge listed buildings datasets from two nations into a single dataset.
+    Args:
+        enhanced_epc_nationone_listed_buildings_df (pl.DataFrame): EPC dataset with listed buildings from nation one
+        enhanced_epc_nationtwo_listed_buildings_df (pl.DataFrame): EPC dataset with listed buildings from nation two
     Returns:
-        pl.DataFrame: Polars DataFrame
+        enhanced_epc_with_all_listed_buildings (pl.DataFrame): EPC dataset with listed buildings from both nations
```

Comment:
```suggestion
        pl.DataFrame: EPC dataset with listed buildings from both nations
```
Can just have the return type without the variable name :)